fileinformer/file_informer.go

```diff
@@ -32,16 +32,34 @@ type Factory struct {
 	// startedInformers is used for tracking which informers have been started.
 	// This allows Start() to be called multiple times safely.
 	startedInformers map[schema.GroupVersionResource]bool
+	opts             FactoryOpts
+}
+
+// FactoryOpts contains the configuration options for the FileInformer factory.
+type FactoryOpts struct {
+	// ResyncPeriod for the shared informer.
+	// Defaults to 1 minute.
+	ResyncPeriod *time.Duration
+}
+
+// fill the default values for the factory options.
+func (f FactoryOpts) fillDefaults() {
+	defaultResync := 1 * time.Minute
+	if f.ResyncPeriod == nil {
+		f.ResyncPeriod = &defaultResync
+	}
 }
 
 var _ dynamicinformer.DynamicSharedInformerFactory = &Factory{}
 
 // NewFileInformerFactory creates a new Factory.
-func NewFileInformerFactory(log logr.Logger) (*Factory, error) {
+func NewFileInformerFactory(log logr.Logger, opts FactoryOpts) (*Factory, error) {
```

Comment:
If we used functional options here, like:

```go
type FactoryOptFunc func(*FactoryOpts)

func WithResyncPeriod(resync time.Duration) FactoryOptFunc {
   return func(opts *FactoryOpts) {
      opts.ResyncPeriod = resync
   }
}

func NewFileInformerFactory(log logr.Logger, ...opts FactoryOptFunc)
```

Then existing call sites would continue working, but you could also pass in a new interval if you needed to:

```go
existingFactory := NewFileInformerFactory(logger)

customResync := NewFileInformerFactory(logger, WithResyncPeriod(5*time.Minute))
```